src/cassandra.go

```diff
@@ -103,6 +101,74 @@ func main() {
 	fatalIfErr(i.Publish())
 }
 
+func runMetricCollection(i *integration.Integration, e *integration.Entity, jmxClient *gojmx.Client) error {
+	if !args.LongRunning {
+		return collectMetrics(e, jmxClient)
+	}
+
+	heartBeat := time.NewTicker(time.Duration(args.HeartbeatInterval) * time.Second)
+	metricInterval := time.NewTicker(time.Millisecond)
+
+	go func() {
+		for range heartBeat.C {
+			log.Debug("Sending heartBeat")
+			// heartbeat signal for long-running integrations
+			// https://docs.newrelic.com/docs/integrations/integrations-sdk/file-specifications/host-integrations-newer-configuration-format#timeout
+			fmt.Println("{}")
+		}
+	}()
+
+	// Force the first collection to happen immediately.
+	first := true
+
+	for range metricInterval.C {
+		if first {
+			metricInterval.Reset(time.Duration(args.Interval) * time.Second)
+			first = false
+		}
+
+		// Check if the nrjmx java sub-process is still alive.
+		if !jmxClient.IsRunning() {
+			return errNRJMXNotRunning
+		}
+
+		e2, err := entity(i)
```

Comment:
Since the entity is only used to collect metrics, would it make sense moving entity creation to `runMetricCollection`? Then, the argument wouldn't be necessary and and we would not need a second integration variable (`e2`).

Reply:
I agree that is weird having to create `e2` when there is already `e` available in the scope 😃